### ISSUE TITLE
Completions integration test is flaky

### ISSUE THREAD

**Issue by minestarks** [microsoft/qsharp#963]
I've seen this test fail about once a week in merge queue checks. Likely some race condition in the test.

Logs:
https://github.com/microsoft/qsharp/actions/runs/7389502439/job/20102481251#step:9:493

On `macos-latest`:

```
%c INFO color: #33f %s%s    Q# Language Service Tests
    Q# Language Service Tests
  GET /static/mount/test.qs?stat 200 5.370 ms - 65
  GET /static/mount/test.qs 200 13.141 ms - 158
  GET /static/build/node_modules/vscode-textmate/release/main.js 200 16.697 ms - 57923
  GET /static/build/node_modules/vscode-oniguruma/release/main.js 200 18.833 ms - 11728
  GET /static/devextensions/language-configuration.json 200 22.019 ms - 403
  %c INFO color: #33f     ✅ %s (%dms) Q# language is registered 91
      ✅ Q# language is registered (91ms)
  GET /static/build/node_modules/vscode-oniguruma/release/onig.wasm 200 21.984 ms - 466610
  GET /static/build/extensions/emmet/dist/browser/emmetBrowserMain.js 200 3.848 ms - 250922
  GET /static/devextensions/out/extension.js 200 39.392 ms - 1313610
  GET /static/devextensions/syntaxes/qsharp.tmLanguage.json 200 1.863 ms - 2274
  GET /static/build/out/vs/base/worker/workerMain.js 200 14.800 ms - 348151
  GET /static/build/out/vs/editor/common/services/editorSimpleWorker.nls.js 200 6.659 ms - 761
  GET /static/build/out/vs/workbench/services/textMate/browser/backgroundTokenization/worker/textMateTokenizationWorker.worker.js 200 3.085 ms - 86489
  GET /static/build/node_modules/vscode-textmate/release/main.js 200 15.927 ms - 57923
  GET /static/build/node_modules/vscode-oniguruma/release/main.js 200 1.885 ms - 11728
  GET /static/build/node_modules/vscode-oniguruma/release/onig.wasm 200 5.261 ms - 466610
  GET /static/devextensions/wasm/qsc_wasm_bg.wasm 200 3.033 ms - 2295989
  %c INFO color: #33f     %d) %s 1 Completions
      1) Completions
  %c INFO color: #33f     ✅ %s (%dms) Definition 1916
      ✅ Definition (1916ms)
  %c INFO color: #33f     ✅ %s Diagnostics
      ✅ Diagnostics
  %c INFO color: #33f     ✅ %s Hover
      ✅ Hover
  %c INFO color: #33f     ✅ %s Signature Help
      ✅ Signature Help
  %c INFO color: #33f %s%s    Q# Notebook Tests
    Q# Notebook Tests
  GET /static/mount/test-no-lang-metadata.ipynb?stat 200 0.834 ms - 65
  GET /static/mount/test-no-lang-metadata.ipynb 200 3.605 ms - 951
  GET /static/build/extensions/ipynb/dist/browser/ipynbMain.js 200 3.047 ms - 166369
  GET /static/build/extensions/python/language-configuration.json 200 4.703 ms - 1573
  GET /static/build/extensions/python/syntaxes/MagicPython.tmLanguage.json 200 4.399 ms - 78248
  %c INFO color: #33f     ✅ %s (%dms) Cell language is set to Q# 831
      ✅ Cell language is set to Q# (831ms)
  GET /static/mount/test.ipynb 200 3.545 ms - 1729
  GET /static/mount/test.ipynb?stat 200 1.644 ms - 66
  %c INFO color: #33f     ✅ %s (%dms) Diagnostics 390
      ✅ Diagnostics (390ms)
  %c INFO color: #33f     ✅ %s Definition
  Error: 1 tests failed.
      ✅ Definition
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:17843:13)
  %c INFO color: #33f   %d passing (%s) 8 5s
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13558:13)
    8 passing (5s)
  	at done2 (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13330:15)
  %c INFO color: #33f   %d failing 1
    1 failing
  %c INFO color: #33f   %s) %s:
       %s
  %s
   1 Q# Language Service Tests
         Completions Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.   	at createTimeoutError$1 (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9205:21)
    	at Runnable$3._timeoutError (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9[496](https://github.com/microsoft/qsharp/actions/runs/7389502439/job/20102481251#step:9:501):16)
    	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9359:32)
    1) Q# Language Service Tests
         Completions:
       Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
    	at createTimeoutError$1 (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9205:21)
    	at Runnable$3._timeoutError (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9496:16)
    	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:9359:32)
  
  %c  ERR color: #f33 Error: 1 tests failed.
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:17843:13)
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13558:13)
  	at done2 (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13330:15)
  Error: 1 tests failed.
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:17843:13)
  	at eval (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13558:13)
  [stored file working copy] handleSaveError(1) - exit - resulted in a save error: Error: Unable to write file 'vscode-test-web://mount/test-no-lang-metadata.ipynb' (Canceled: Canceled) vscode-test-web://mount/test-no-lang-metadata.ipynb notebook/jupyter-notebook
  	at done2 (http://localhost:3000/static/devextensions/test/out/language-service/index.js#vscode-extension:13330:15)
  %c  ERR color: #f33 [stored file working copy] handleSaveError(1) - exit - resulted in a save error: Error: Unable to write file 'vscode-test-web://mount/test-no-lang-metadata.ipynb' (Canceled: Canceled) vscode-test-web://mount/test-no-lang-metadata.ipynb notebook/jupyter-notebook
  Failed to save 'test-no-lang-metadata.ipynb': Unable to write file 'vscode-test-web://mount/test-no-lang-metadata.ipynb' (Canceled: Canceled)
  Failed to run tests Error: Test failed
      at /Users/runner/work/qsharp/qsharp/node_modules/@vscode/test-web/out/index.js:59:23
      at async BindingCall.call (/Users/runner/work/qsharp/qsharp/node_modules/playwright-core/lib/client/page.js:634:130)
      at async Page._onBinding (/Users/runner/work/qsharp/qsharp/node_modules/playwright-core/lib/client/page.js:161:7)
  npm ERR! Lifecycle script `test` failed with error: 
  npm ERR! Error: command failed 
  npm ERR!   in workspace: qsharp-lang-vscode-dev@0.0.0 
  npm ERR!   at location: /Users/runner/work/qsharp/qsharp/vscode 
  Traceback (most recent call last):
    File "/Users/runner/work/qsharp/qsharp/./build.py", line 414, in <module>
      subprocess.run(vscode_args, check=True, text=True, cwd=vscode_src)
    File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
      raise CalledProcessError(retcode, process.args,
  subprocess.CalledProcessError: Command '['npm', 'test', '--', '--verbose']' returned non-zero exit status 1.
  Error: Process completed with exit code 1.
```

**Comment by minestarks** [comment]
I think this has to do with `activate()` taking too long, though I'm unable to confirm. I tried about a dozen runs today with very verbose logging, but could not hit the timeout. Seems to just happen more often on some days than others.

**Comment by swernli** [comment]
We've seen consistent behavior in integration tests since the fix in #1015, so I think it's safe to close this now.
